src/cpp/transport/TCPTransportInterface.cpp

```diff
@@ -1445,33 +1168,35 @@ void TCPTransportInterface::SecureSocketAccepted(
 }
 #endif
 
+//TODO Passh the shared_ptr
```

Comment:
This is a very important TODO, especially with security. It should be implemented before merging.